src/include/duckdb/planner/table_binding.hpp

```diff
@@ -132,7 +132,7 @@ struct TableBinding : public Binding {
 	const vector<ColumnIndex> &GetBoundColumnIds() const;
 
 protected:
-	ColumnBinding GetColumnBinding(column_t column_index);
+	ColumnBinding GetColumnBinding(column_t column_index, const LogicalType &type);
```

Comment:
Forward the type of the column here, so we can create ColumnIndex's with the correct base type